src/persist/src/indexed/background.rs

```diff
@@ -289,21 +298,21 @@ mod tests {
         let expected_res = CompactTraceRes {
             req: req.clone(),
             merged: TraceBatchMeta {
-                key: "MERGED_KEY".into(),
+                keys: vec!["MERGED_KEY".into()],
                 format: ProtoBatchFormat::ParquetKvtd,
                 desc: desc_from(0, 3, 2),
                 level: 1,
                 size_bytes: 0,
             },
         };
         let mut res = maintainer.compact_trace(req).recv()?;
-        let merged_key = res.merged.key.clone();
-        res.merged.key = "MERGED_KEY".into();
+        let merged_key = res.merged.keys.clone();
+        res.merged.keys = vec!["MERGED_KEY".into()];
         res.merged.size_bytes = 0;
         assert_eq!(res, expected_res);
 
         let b2 = blob
-            .get_trace_batch_async(&merged_key, CacheHint::MaybeAdd)
+            .get_trace_batch_async(&merged_key[0], CacheHint::MaybeAdd)
```

Comment:
No I think we changed res.merged_keys to `vec![MERGED_KEY]` so that we could assert on res, but we need the original list of keys (before this change the original key) to get the actual batch. When I make the change that was requested, Blob complains that no key named `MERGED_KEY` exists in the blob.

Reply:
okay, sure, ignore that specific suggestion then. to clarify my original comment, I think it's possible to avoid the [0], let's figure out a way to do that

Reply:
done!